tmol/tests/pose/compiled/test_all_pairs_shortest_paths.py

```diff
@@ -94,8 +100,44 @@ def test_all_pairs_shortest_paths_w_off_diagonal_bonds():
 
     weights_cuda = weights.clone().to(torch_device_cuda)
 
-    stacked_apsp(weights)
-    stacked_apsp(weights_cuda)
+    stacked_apsp(weights, -1)
+    stacked_apsp(weights_cuda, -1)
+
+    torch.testing.assert_close(weights, weights_cuda.cpu())
+
+    cuda_result_cpu = weights_cuda.cpu().numpy()
+    for i in range(n_nodes):
+        for j in range(n_nodes):
+            assert cuda_result_cpu[0, i, j] == cuda_result_cpu[0, j, i]
+
+
+@requires_cuda
+def test_all_pairs_shortest_paths_w_off_diagonal_bonds_and_threshold():
+    # note that there is an expected difference in behavior here:
+    # the cuda implementation does not respect the threshold and
+    # so reports the actual path distance; it has to be "corrected"
+    # to match the CPU's computed result
+    torch_device_cpu = torch.device("cpu")
+    torch_device_cuda = torch.device("cuda")
+
+    n_nodes = 128
+    n_graphs = 5
+
+    weights = torch.full(
+        (n_graphs, n_nodes, n_nodes), -1, dtype=torch.int32, device=torch_device_cpu
+    )
+    arange_n_nodes = torch.arange(n_nodes, dtype=torch.int64, device=torch_device_cpu)
+    weights[:, arange_n_nodes, arange_n_nodes] = 0
+    weights[:, arange_n_nodes[1:], arange_n_nodes[:-1]] = 1
+    weights[:, arange_n_nodes[:-1], arange_n_nodes[1:]] = 1
+    weights[:, 10, 20] = 1
+    weights[:, 20, 10] = 1
+
+    weights_cuda = weights.clone().to(torch_device_cuda)
+
+    stacked_apsp(weights, 6)
+    stacked_apsp(weights_cuda, 6)
+    weights_cuda[weights_cuda > 6] = 6
```

Comment:
presumably cuda 'stacked_apsp' does not use this max count.  I think line 140 should be included in the cuda implementation ... we want this function to behave identically on CPU and CUDA.